### DISCUSSION TITLE
Add additional filters based on another filters value

### DISCUSSION THREAD

**Discussion by tobiashammarstrand** [rappasoft/laravel-livewire-tables#1310] [Q&A] [answered]
Hi Community,

I would like to have a "subfilter" that is shown to the user when the user selects the "main" filter. In the application, there are dependent columns shown depending on selected values on the main filter. So I would like to also be able to add dependent filters.

How could I trigger rerender of the filters? Or set the filter value before filters are rendered?

Thanks!

**Comment by lrljoe** [comment, marked as answer]
You can set the filter value using the setFilter() approach.

At the moment, you'd have to trigger a re-render of the whole component to get the filters to re-render.

Improved filter methods are on the way for 3.x+ (starting with 2 additional filter types) though!
